.scripts/_tmux.sh

```
@@ -0,0 +1,259 @@
+}
+
+# Set up trap to catch interrupt and clean up resources
+# Arguments:
+#   $1: exit_session (boolean, optional) - Whether to exit the tmux session on cleanup
+# Returns:
+#   None
+function tmux:setup_trap() {
+  local exit_session="${1:-true}"
+  
+  # Create a trap function that calls our cleanup
+  trap "tmux:cleanup_all $exit_session" INT TERM EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)